sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java

```diff
@@ -18,22 +18,60 @@
 
 package org.sormas.e2etests.steps.web.application.contacts;
 
-import static org.sormas.e2etests.pages.application.contacts.ContactDirectoryPage.NEW_CONTACT_BUTTON;
+import static org.sormas.e2etests.pages.application.contacts.ContactDirectoryPage.*;
 import static org.sormas.e2etests.pages.application.contacts.CreateNewContactPage.FIRST_NAME_OF_CONTACT_PERSON_INPUT;
+import static org.sormas.e2etests.pages.application.contacts.EditContactPage.UUID_INPUT;
 
+import com.google.common.truth.Truth;
 import cucumber.api.java8.En;
 import javax.inject.Inject;
+import org.openqa.selenium.By;
 import org.sormas.e2etests.helpers.WebDriverHelpers;
+import org.sormas.e2etests.state.ApiState;
+import org.sormas.e2etests.steps.BaseSteps;
 
 public class ContactDirectorySteps implements En {
 
-  @Inject
-  public ContactDirectorySteps(WebDriverHelpers webDriverHelpers) {
+  protected BaseSteps baseSteps;
+  protected WebDriverHelpers webDriverHelpers;
 
+  @Inject
+  public ContactDirectorySteps(
+      WebDriverHelpers webDriverHelpers, BaseSteps baseSteps, ApiState apiState) {
+    this.baseSteps = baseSteps;
+    this.webDriverHelpers = webDriverHelpers;
     When(
         "I click on the NEW CONTACT button",
         () ->
             webDriverHelpers.clickWhileOtherButtonIsDisplayed(
                 NEW_CONTACT_BUTTON, FIRST_NAME_OF_CONTACT_PERSON_INPUT));
+
+    When(
+        "I open the last created contact",
+        () -> {
+          searchAfterContactByMultipleOptions(apiState.getCreatedContact().getUuid());
+          openContactFromResultsByUUID(apiState.getCreatedContact().getUuid());
+        });
+
+    Then(
+        "I check that number of displayed contact results is {int}",
+        (Integer number) -> {
+          Truth.assertThat(webDriverHelpers.getNumberOfElements(CONTACT_GRID_RESULTS_ROWS))
```

Comment:
Please use this inside of `assertHelpers.assertWithPoll` methods to have a async assert